frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/templates/assisted-installer/ai-template.hbs

```diff
@@ -39,16 +39,25 @@ spec:
     name: '{{{ai.name}}}'
   holdInstallation: true
   provisionRequirements:
-    workerAgents: 0
-{{#switch ai.highAvailabilityMode}}
-  {{#case 'Full'}}
-    controlPlaneAgents: 3
-  {{/case}}
-
-  {{#case 'None'}}
+    workerAgents: 0    
+    {{#if ai.controlPlaneAgents}}
```

Comment:
no need for switch statement. A simple
```suggestion
    workerAgents: 0    
    controlPlaneAgents: {{{ai.controlPlaneAgents}}}
```

will do

Reply:
the problem is that controlPlaneAgents is an string and we need a number

Reply:
after fixing https://github.com/openshift-assisted/assisted-installer-ui/pull/2727, we will have a number here.

also `cim-template.hbs` needs to be updated.

also in order to make the  yaml editor -> UI form sync work, we need to update https://github.com/stolostron/console/blob/main/frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/DetailsForm.tsx#L138-L151

by adding the `controlPlaneCount` 
```
controlPlaneCount: { path: 'AgentClusterInstall[0].spec.provisionRequirements.controlPlaneAgents' },
```